pkg/controller/controlplane/valuesprovider.go

```diff
@@ -720,6 +720,12 @@ func getControlPlaneShootChartValues(
 		}
 	}
 
+	if value, ok := cluster.Shoot.Annotations[aws.legacyXFS]; ok {
+		csiDriverNodeValues["driver"] = map[string]interface{}{
+			"legacyXFS": value,
+		}
+	}
+
```

Comment:
With the code as it is, you would overwrite the value of `VolumeAttachLimit` in the `driver` key of the map. You should only create the map if it's not there.